ppl/zqd/ingest/multipartlogreader.go

```diff
@@ -46,6 +53,12 @@ read:
 		if zr == nil || err != nil {
 			return nil, err
 		}
+		if m.shaperAST != nil {
+			zr, err = driver.NewReader(context.Background(), m.shaperAST, m.zctx, zr)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Consider moving this into the `next` function that creates the reader.

Reply:
done in 6a76767624ade2687273a0b2d3dbeef14d2a7673